### ISSUE TITLE
SVGs with no viewBox and width&height attribs

### ISSUE THREAD

**Issue by mgrudzinska** [thorvg/thorvg#1239]
I need to read what the expected behavior is, but for sure something has to be rendered (there is not for now)
pointed out in https://github.com/godotengine/godot/pull/64530

**Comment by bruvzg** [comment]
Some extra info:

Here are a few examples of the SVG code extracted from fonts, it's either missing `viewBox` completely or it's not enclosing the image:

`🍏` from Twitter Color Emoji font:

```svg
<svg xmlns="http://www.w3.org/2000/svg" id="glyph318"><g transform="translate(0 -6.75) translate(0,-1638.4) scale(56.888888888888886)"><path fill="#77B255" d="M24 7c-3 0-3 1-6 1s-3-1-6-1c-4 0-9 2-9 9 0 11 6 20 10 20 3 0 3-1 5-1s2 1 5 1c4 0 10-9 10-20 0-7.001-5-9-9-9z" /><path fill="#3E721D" d="M17.311 7.88s-1.775-4.674-6.58-6.06c-3.843-1.108-6.318.26-6.318.26s3.012 3.991 5.895 4.822c2.882.83 7.003.978 7.003.978z" /><path fill="#662113" d="M18 10a1 1 0 0 1-1-1c0-3.441 1.2-6.615 3.293-8.707a.999.999 0 1 1 1.414 1.414C19.986 3.427 19 6.085 19 9a1 1 0 0 1-1 1z" /></g></svg>
```

<img width="200" alt="Screenshot 2022-08-20 at 10 20 18" src="https://user-images.githubusercontent.com/7645683/185733991-f0a49ce8-401c-47cb-9de9-04e9b5dba66c.png">


`🍏` from EmojiOne Color font:

```svg
<?xml version='1.0' encoding='utf8'?>
<svg xmlns="http://www.w3.org/2000/svg" id="glyph310" viewBox="0 0 64 64"><g transform="translate(0 -9.6) translate(-0.0 -104.8576) scale(2.048)"><path d="M31.4 4.9c1.1 2.5-.5 10.9-.5 10.9s-7.5-4.4-8.6-6.9c-1.1-2.5 0-5.4 2.5-6.5 2.5-1.1 5.5 0 6.6 2.5" fill="#64892f" /><path d="M33.3 16.3h-2.6c0-4.3 2.4-11.6 12-11.6v2.6c-8.7 0-9.4 8.2-9.4 9" fill="#947151" /><path d="M32 15.8C24.7 8.4 3 8.1 3 29.3 3 42.7 18.8 62 25.7 62c3.1 0 4.2-2 6.1-2 1.8 0 2.4 1.9 6.4 1.9C45.1 62 61 42.8 61 29.3 61 8.1 39.3 8.4 32 15.8z" fill="#8cc63e" /></g></svg>
```

<img width="200" alt="Screenshot 2022-08-20 at 10 18 32" src="https://user-images.githubusercontent.com/7645683/185733997-4118e4db-a336-4154-b3d3-44d9c36d89f7.png">

A sample SVG font rendering code, using librsvg: https://gitlab.freedesktop.org/freetype/freetype-demos/-/blob/master/src/rsvg-port.c

- `viewBox` / `width, height` is only used to determine scaling factors - lines `225 ... 256`
- Only element with ID equal to `glyph<ID>` is rendered, using its actual bounding box, document `viewBox` / `width, height` are ignored - lines `303 ... 333`

**Comment by mgrudzinska** [comment]
@bruvzg thx for your comment.
One thing is how to deal without any viewBox and viewPort information. 
But the other one is to deal with examples that you gave. Please correct me if I'm wrong, but according to the svg w3c standard, the svg from EmojiOne Color font should be drawn exactly as it is - so one can see only a small part of the apple. 

There is a comment in the librsvg:
"TODO: I'm not sure what the standard says about the situation if
`ViewBox` as well as `width`/`height` are present; however, I've
never seen that situation in real fonts."
I know what the svg standard says, doesn't it apply to fonts/emoji? 
I'm not familiar with [this ](https://www.w3.org/TR/css-fonts-4/) document - just scrolled and I didn't find any special info about viewBox in svgs. Do you know something more? Some documentation/standard, that I should check?

So probably we could treat viewbox/widht&height info in some other way for emoji, but will it be correct for all of the files?

**Comment by bruvzg** [comment]
Relevant part of OpenType specification - https://docs.microsoft.com/en-us/typography/opentype/spec/svg#coordinate-systems-and-glyph-metrics

`viewBox` / `widht` / `height` are used only to determine scaling, if it's not present `unitsPerEm` is used as a viewport size.

> But the other one is to deal with examples that you gave. Please correct me if I'm wrong, but according to the svg w3c standard, the svg from EmojiOne Color font should be drawn exactly as it is - so one can see only a small part of the apple.

It should not be clipped, I guess it's valid for plain SVG, but not for OpenType.

> Although the initial viewport size is the em square, the viewport must not be clipped. The <svg> element is assumed to have a clip property value of auto, and an overflow property value of visible. A font should not specify clip or overflow properties on the <svg> element. If clip or overflow properties are specified on the <svg> element with any other values, they must be ignored.

**Comment by hermet** [comment]
@mgrudzinska close?

**Comment by mgrudzinska** [comment]
yes! 
solved in #1318 with fix in #1344
